launch_handler-explainer.md

```diff
@@ -0,0 +1,141 @@
+# Web App Launch Handling
+
+## Overview
+
+This document describes a new `launch_handler` manifest member that enables
+web apps to customise their launch behaviour across all types of app launch
+triggers.
+
+
+## Use Cases
+
+- Web apps that are designed to be used in a single window e.g. a music app.
+
+- Web app that capture and handle share target events and user navigations
+  in existing windows without invoking a navigation and losing existing state.
+  E.g. sharing an image to a chat web app could open a picker overlayed on top
+  of an existing conversation to select which contact to send it to.
+
+
+## Background
+
+There are several ways for a web app window to be opened:
+- Platform specific app launch surface
+- [File handling](https://github.com/WICG/file-handling/blob/main/explainer.md)
+- [Share target](https://w3c.github.io/web-share-target/)
+- [In scope link capturing](https://github.com/WICG/sw-launch/blob/master/declarative_link_capturing.md)
+- [Shortcuts](https://www.w3.org/TR/appmanifest/#dfn-shortcuts)
+- [Note taking](https://wicg.github.io/manifest-incubations/index.html#note_taking-member)
+- [Protocol handling](https://github.com/MicrosoftEdge/MSEdgeExplainers/blob/main/URLProtocolHandler/explainer.md)
+
+Web apps launched via these triggers will open in a new or existing app window
+depending on the user agent platform. There is currently no mechanism for the
+web app to configure this behaviour.
+
+
+## Proposal
+
+- Add a `launch_handler` member to the web app manifest specifying the default
+  client selection and navigation behaviour for web app launches.
+  The shape of this member is as follows:
+  ```
+  "launch_handler": {
+    "route_to_client": null | "new" | "existing",
+    "navigate_client": null | true | false
```

Comment:
I don't like that the names are so similar, and it is quite hard to notice (from the name) that navigate_client is a boolean

Reply:
This was originally `"route_to": null | "new-client" | "existing-client"`, happy to go with that instead.

Reply:
We use "auto" as 'up to the user agent to decide' elsewhere: https://www.w3.org/TR/appmanifest/#ref-for-dfn-auto-1

@marcoscaceres

Reply:
I personally find `"auto" | "new-client" | "existing-client"` easier to reason about

Reply:
If it is undefined, the default fallback value would be "auto"

Reply:
Thanks for the precedent link, updated.